src/agent/src/device.rs

```diff
@@ -53,14 +53,42 @@ pub const DRIVER_VFIO_GK_TYPE: &str = "vfio-gk";
 // container as a VFIO device node
 pub const DRIVER_VFIO_TYPE: &str = "vfio";
 
-#[derive(Debug)]
-struct DevIndexEntry {
-    idx: usize,
-    residx: Vec<usize>,
+#[derive(Debug, Clone)]
+struct DevSpecFixup {
+    // the path to the device in the original OCI spec
+    container: String,
```

Comment:
I'm still slightly confused by these names tbh. How about:

- `container` => `oci_spec_path`
- `vm` => `vm_path` (or `container_path`)
- `update` => `update_path`

Reply:
TBH, I'm really struggling to find good names for these amongst other things.  (`container_path` for the thing that's currently `vm` is definitely wrong though; that path might not appear within the container in any form).

Reply:
@dgibson - haha, yes names are always surprisingly hard I find. Anyone else have thoughts on the above?

Reply:
Well, I've done a substantial rework for the next spin.  I think it makes these clearer, but let's see what you think.